player/playloop.c

```diff
@@ -258,6 +261,15 @@ void reset_playback_state(struct MPContext *mpctx)
     update_core_idle_state(mpctx);
 }
 
+static double calculate_framestep_pts(MPContext *mpctx, double current_time,
+                                      int step_frames)
+{
+    // Crude guess at the pts.
+    int previous_frame = mpctx->num_past_frames - 1;
+    double pts = mpctx->past_frames[previous_frame].approx_duration * step_frames;
```

Comment:
This looks like it falls apart instantly with VFR

I think I prefer if `frame-step` is 'really slow' rather than 'normal slow but sometimes it doesn't work at all'

Reply:
For going forward, I could let it just play X amount of frames then stop. This is how I implemented at first, but that seemed pretty unintuitive to me personally (if you want to skip a large amount of frames forward, you're basically just watching the video until it arrives). It would be more accurate though. Maybe I could add an additional flag for that. When going backwards, I don't think there's a reasonable solution. `frame-back-step` should already be pretty unreliable in that case since it just does a seek to a specific pts.

Well, I guess you could try playing backwards X amount of frames. Dunno if we really want to mess with that.